src/ensembl/src/content/app/custom-download/components/checkbox-grid/CheckboxGrid.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+
+import styles from './CheckBoxGrid.scss';
+
+type Props = {
+  gridData: any;
+  columns: number;
+  hideUnchecked?: boolean;
+  hideTitles?: boolean;
+  checkboxOnChange: (status: boolean, subSection: string, id: string) => void;
+};
+
+export const filterCheckedAttributes = (attributes: any) => {
+  const filteredAttributes: any = {};
+
+  Object.keys(attributes).forEach((section) => {
+    Object.keys(attributes[section]).forEach((attributeId) => {
+      if (attributes[section][attributeId].checkedStatus === true) {
+        if (!filteredAttributes[section]) {
+          filteredAttributes[section] = {};
+        }
+        filteredAttributes[section][attributeId] =
+          attributes[section][attributeId];
+      }
+    });
+  });
+
+  return filteredAttributes;
+};
+
+export const getAttributesCount = (attributes: any) => {
+  let totalAttributes = 0;
+
+  if (!attributes || Object.keys(attributes).length === 0) {
+    return 0;
+  }
+  Object.keys(attributes).forEach((section) => {
+    totalAttributes += Object.keys(attributes[section]).length;
+  });
+
+  return totalAttributes;
+};
+
+const renderCheckBoxList = (
+  checkboxList: any,
+  props: Props,
+  subSection: string
+) => {
+  if (!Object.keys(checkboxList).length) {
+    return null;
+  }
+
+  const checkboxListIDs = Object.keys(checkboxList).sort();
+
+  const gridMatrix = Array(props.columns).fill(0);
+
+  let totalCheckbox = checkboxListIDs.length;
+
+  for (let i = 0; i < checkboxListIDs.length; i++) {
+    if (totalCheckbox <= 0) {
+      break;
+    }
+    for (let j = 0; j < props.columns; j++) {
+      if (totalCheckbox <= 0) {
+        break;
+      }
+      totalCheckbox -= 1;
+      if (checkboxListIDs[i + j]) {
+        gridMatrix[j] += 1;
+      }
+    }
+  }
```

Comment:
Is this loop the same as saying:

```
const rows = Math.ceil(checkboxListIDs / props.columns);
const gridMatrix = Array(props.columns).fill(rows);
```
(and, judging by how the `gridMatrix` is used below, you probably don't even need it):

```
import range from 'lodash/range';

...

<div className={styles.checkboxGridContainer}>
  { range(0, columns).map((column) => ...)}
</div>
```

Reply:
As discussed, range cannot be used here as the structure of the gridMatrix is slightly different.
We can get it replaced once we have a simpler logic.